### ISSUE TITLE
TestApp generation should add necessary imports

### ISSUE THREAD

**Issue by yulvil** [elliotchance/c2go#267]
Function TestApp does make use of `os`, `ioutil`, `noarch` and `testing` packages. We should make sure that those imports are always included.

```
import "io/ioutil"
import "os"
import "testing"
import "github.com/elliotchance/c2go/noarch"

func TestApp(t *testing.T) {
   os.Chdir("../../..")
   ioutil.WriteFile("build/stdin", []byte{'7'}, 0777)
   stdin, _ := os.Open("build/stdin")
   noarch.Stdin = noarch.NewFile(stdin)
   main()
}
```

It is possible to create trivial programs where the `main` function do not refer to those imports.
```
int main() {
    return 97;
}
```
=> `./main.go:21:2: undefined: noarch`

Low priority since any non-trivial C program will import "os" and "noarch".

**Comment by elliotchance** [comment]
`TestApp` should only be generated under `go test`. Are you seeing this output from `c2go transpile`? If you are then this is a bug and `OutputAsTest` is not `false` (the default mode): https://github.com/elliotchance/c2go/blob/master/program/program.go#L76

**Comment by yulvil** [comment]
@elliotchance Yes I am seeing this with `go transpile`. I will look into this later.

**Comment by yulvil** [comment]
I confirm that `p.OutputAsTest = true // args.outputAsTest` in main.c

https://github.com/elliotchance/c2go/blame/master/main.go#L227

**Comment by elliotchance** [comment]
That was a bug introduced by myself... You can even see the original code commented out. 😢

If you have time please change that line back to:

```go
p.OutputAsTest = args.outputAsTest
```

That will also fix this ticket.

**Comment by yulvil** [comment]
TODO: Verify that this is fixed by 1a2f0890b3e2dd1727b498ca3d6c5e9169ec4bab

**Comment by elliotchance** [comment]
I can confirm that it was *not* fixed in that PR (already merged).
